### PR TITLE
 Some of the features include

### PR DESCRIPTION
**Pull request by ldecicco-USGS**
(a) artifacting the R package installation library and using that for each of the future stages (b) installing a github package (c) checking the package (d) running unit tests (e) checking style (f) checking code coverage (g) adding the check/test output as artifacts that can be downloaded if needed

### CODE REVIEW

File: gitlab-ci/wateruse.gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    - mkdir -p $APT_CACHE
+    - apt-get update
+    - apt-get -o dir::cache::archives=${APT_CACHE} install -y --no-install-recommends ${APT_PKGS}
+
+getready:
+  stage: .pre
+  script:
+    - mkdir -p $R_LIBS_USER
+    - R -e 'install.packages(c("lintr","ggthemes", "sp", "mapproj", "maps", "maptools", "rgeos", "shinycssloaders", "shinydashboard"), repos = Sys.getenv("CRAN_MIRROR"), lib = Sys.getenv("R_LIBS_USER"))'
+    - R -e 'withr::with_libpaths(new = Sys.getenv("R_LIBS_USER"), remotes::install_github("USGS-R/wateRuse",  repos = Sys.getenv("CRAN_MIRROR")))'
```
Comment (MichaelChirico) [inline gitlab-ci/wateruse.gitlab-ci.yml:28]: is this package used?

Comment (MichaelChirico) [reply gitlab-ci/wateruse.gitlab-ci.yml:28]: oh! i didn't read the file name.

**Comment by MichaelChirico** [comment]
Is this file hosted on GitHub or GitLab?

if GitHub, you can add the repo to `.tracked_repos` file and it will automatically update.

if GitLab, can you tell me the repo and I'll try and adjust the update script to do this

**Comment by ldecicco-USGS** [comment]
It's a private gitlab repo....but now that you mention it, I do have a similar public one that I can use. I can close this PR and open a new one soon that'll have a repo to link to.

**Comment by MichaelChirico** [comment]
ok great! for the record, I don't see a problem per se w tracking a file from a private repos, just that updates will have to be manual :)

**Pull request closed without merging**